contracts/smart-contract-wallet/base/FallbackManager.sol

```diff
@@ -51,5 +33,29 @@ contract FallbackManager is SelfAuthorized {
         }
     }
 
+    /// @dev Allows to add a contract to handle fallback calls.
+    ///      Only fallback calls without value and with data will be forwarded.
+    ///      This can only be done via a Safe transaction.
+    /// @param handler contract to handle fallback calls.
+    function setFallbackHandler(address handler) public authorized {
+        // review - check if this is loading the correct slot, for previousHandler indexing
+        address previousHandler;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            previousHandler := sload(FALLBACK_HANDLER_STORAGE_SLOT)
+        }
+        _setFallbackHandler(handler);
+        emit ChangedFallbackHandler(previousHandler, handler);
```

Comment:
verify if this is loading the correct slot, for previousHandler indexing

Reply:
this looks fine to me. it will load and store it in the variable, then it calls internal method that updates storage slot so previousHandler would be accurate. can write the test case